### ISSUE TITLE
Discuss how to get notification permission during app usage

### ISSUE THREAD

**Issue by gabelula** [ooni/probe#813]
Ask user notification permission during app usage

**Comment by elioqoshi** [comment]
When will this happen for the first time? We should leave it rather late if possible before the first notification would appear, so that users wouldn't feel overwhelmed.

**Comment by hellais** [comment]
@lorenzoPrimi What is the current status of this? Do we still not offer any modal to tell people to enable notifications?

**Comment by lorenzoPrimi** [comment]
Exactly. We actually were waiting to have some features to offer such as news or orchestration, or even the ooni run by push notification.

**Comment by lorenzoPrimi** [comment]
@hellais why did you put this into sprint 5? If you want to keep it here please write exact specs on how it should be implemented

**Comment by hellais** [comment]
I put it in this sprint with the label `discuss`, to highlight that we should think about how this feature works and if there is some way we can get some easy way of doing it, since we are to some extent reliant on push notifications for informing users of targets for testing.

**Comment by lorenzoPrimi** [comment]
Ok understood,  so it can stay there

**Comment by lorenzoPrimi** [comment]
@hellais according to app logic I can show the "pre"notification popup from anywhere in the app.
The discussions should be about when and why.

**Comment by bassosimone** [comment]
We agreed to add a modal that appears after the user runs the first test saying something like “thanks for running OONI probe, can we count on you to run tests?”

And then a second screen saying that we will request notification permissions, “are you okay with that?”.

It they say no, we will retrigger the modal every X time after they run a test and maybe offer a way for dismissing the dialogue forever.

@hellais will work on a mock-up and @agrabeli will write the copy.

**Comment by lorenzoPrimi** [comment]
As we don't have any orchestration in the app this popup would be more to send notification to the user (news, run this list etc)

**Comment by bassosimone** [comment]
@lorenzoPrimi correct!

**Comment by lorenzoPrimi** [comment]
We still need to answer two main questions
- When to show the popup? After the first test ran? the second? At the app opening? When to repeat it? 
There are multiple strategies and we need to tailor this in an optimised way.
For example iRate https://github.com/nicklockwood/iRate, a component used to ask to rate the app in the app store offer these options we can get inspiration from:
- daysUntilPrompt
- usesUntilPrompt
- eventsUntilPrompt
- remindPeriod

The second issue  is about the copy, I'll leave that to @agrabeli

**Comment by hellais** [comment]
We should discuss this during the next sprint and come up with a plan.

**Comment by lorenzoPrimi** [comment]
@hellais can you summarize what it should be implemented based on the input you got?

**Comment by hellais** [comment]
We did not discuss this during the frontend meetings. Let's try and speak about it during the next one.

**Comment by hellais** [comment]
I started doing some research on this topic in this document: https://docs.google.com/document/d/1x1jMjMq4-e8Xb4USegBSUV6V_cpNUWb2GwfBE5jhAV4/edit#

**Comment by hellais** [comment]
We have a document outlining some research and solutions. We can consider the discussion done. We should open a new issue to review when we are closer to implementation.
